### ISSUE TITLE
There is no way to select source for recording audio

### ISSUE THREAD

**Issue by malaniz** [expo/expo#9012]
### Summary of Issue (just a few sentences)
There is no way to select source for recording audio
https://github.com/expo/expo/blob/d91380e2f6ae697102b7e0a279b06507c82a411c/packages/expo-av/android/src/main/java/expo/modules/av/AVManager.java#L605
DEFAULT is always used.

### Reproducible Demo

- This should include as little code as possible, please don't simply link your entire project
- Sharing a link to a [Snack](https://snack.expo.io/) is a GREAT way to provide a reproducible demo :) 
- If a reproducible demo, or a complete list of steps from blank project to bug, are not provided, it is very likely your issue will be closed
- If you need more guidance, please see https://stackoverflow.com/help/mcve

As an added benefit- creating a repro may help you identify the source of the bug, which means we are one step closer to fixing it! Thanks for helping us help you!

**Comment by byCedric** [comment]
Hi @malaniz! It looks like we don't have an API in place for this (yet). The best thing you can do to get us working on that is either open a [feature request on canny (and promote it)](https://expo.canny.io/), or open a PR with a draft implementation of the API you mentioned. There are a lot of things we should improve on the AV/Audio/Video APIs, but these are blocked by the [TurboModules integration](https://github.com/expo/expo/issues/4451) (if I'm correct, cc @sjchmiela).

Hope it helps!

**Comment by malaniz** [comment]
@byCedric thanks! I'll try to make a PR with a solution. I didn't find another way to select a different source to record my audio. If you know how or someone know would be great!, thanks

**Comment by LudoLamerre** [comment]
Yes this feature is very expected because it is impossible to make applications using the microphone so far because when using the bluetooth helmet the recording is made via the phone microphone...
I make two applications with recording features and I need to not used expo because of that.
